populator/nosql/couchbase/src/main/java/org/jboss/arquillian/populator/nosql/couchbase/CouchbaseOptions.java

```diff
@@ -0,0 +1,112 @@
+package org.jboss.arquillian.populator.nosql.couchbase;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class taht represents custom options for Couchbase
+ */
+public class CouchbaseOptions implements Map<String, Object> {
+
+    static String BUCKET_PASSWORD = "BUCKET_PASSWORD";
+    static String CREATE_BUCKET = "CREATE_BUCKET";
+    static String CLUSTER_USERNAME = "CLUSTER_USERNAME";
+    static String CLUSTER_PASSWORD = "CLUSTER_PASSWORD";
+
+    private Map<String, Object> options = new HashMap<>();
+
+    private CouchbaseOptions() {
+    }
+
+    public static CouchbaseConfigurationOptions options() {
+        return new CouchbaseConfigurationOptions();
+    }
+
+    @Override
+    public int size() {
+        return options.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
```

Comment:
We implement a map because `withOptions` https://github.com/arquillian/arquillian-extension-persistence/pull/61/files#diff-6a6b351b5f8b6bccba61feceb94b0fb1R82 must be generic enough to fit any NoSQL database, Each one might have different attributes. So you've got a map so developer can set it in a generic way (properties file) or by using the custom builder we offer for each database.